consensus/src/network_tests.rs

```diff
@@ -318,34 +386,80 @@ impl NetworkPlayground {
 
     pub async fn start<T: Payload>(mut self) {
         // Take the next queued message
-        while let Some((src, net_req)) = self.outbound_msgs_rx.next().await {
-            // Deliver and copy message it if it's not dropped
-            if !self.is_message_dropped(&src, &net_req) {
-                self.deliver_message::<T>(src, net_req).await;
+        while let Some((src_twin_id, net_req)) = self.outbound_msgs_rx.next().await {
+            // extract destination peer
+            let dst = match &net_req {
+                PeerManagerRequest::SendMessage(dst, _) => *dst,
+                msg => panic!(
+                    "[network playground] Unexpected PeerManagerRequest: {:?}",
+                    msg
+                ),
+            };
+
+            let dst_twin_ids = self.get_twin_ids(dst);
+
+            for dst_twin_id in dst_twin_ids.iter() {
+                // convert PeerManagerRequest to corresponding PeerManagerNotification
+                let msg_notif = match &net_req {
```

Comment:
don't need to repeat this pattern match since we already have one above.
```
let (dst, msg) = match net_req {
};

let msg_notif = PeerManagerNotification::RecvMessage(src_twin_id.author, msg.clone())
```

Reply:
Fixed.